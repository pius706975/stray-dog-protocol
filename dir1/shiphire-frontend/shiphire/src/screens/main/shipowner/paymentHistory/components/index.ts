export { default as HistoryList } from './HistoryList';
export { default as ModalImage } from './ModalImage';
