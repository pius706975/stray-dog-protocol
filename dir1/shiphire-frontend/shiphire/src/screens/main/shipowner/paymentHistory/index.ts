export { default as PaymentOwnerHistory } from './PaymentOwnerHistory';
