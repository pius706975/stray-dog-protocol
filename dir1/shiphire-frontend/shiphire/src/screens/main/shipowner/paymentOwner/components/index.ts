export { default as SelectedImageAccordion } from './SelectedImageAccordion';
export { default as HistoryCard } from './HistoryCard';
