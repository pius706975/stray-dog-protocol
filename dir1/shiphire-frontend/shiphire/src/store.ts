import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
