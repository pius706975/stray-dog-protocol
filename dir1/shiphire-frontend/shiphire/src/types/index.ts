export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'