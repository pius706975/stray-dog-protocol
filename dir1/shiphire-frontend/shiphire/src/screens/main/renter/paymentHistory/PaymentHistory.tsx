import { Text, FlatList } from 'react-native';
import React from 'react';
import { Payment, PaymentHistoryProps } from '../../../../types';
import { HistoryList } from './components';
import { useGetPaymentHistory } from '../../../../hooks';
import { handleAxiosError } from '../../../../utils';
import { View } from 'react-native-ui-lib';
import { Color } from '../../../../configs';
import { useIsFocused } from '@react-navigation/native';

const PaymentHistory: React.FC<PaymentHistoryProps> = ({ route }) => {
    const { rentalId } = route.params;
    const mutationGetPaymentHistory = useGetPaymentHistory();
    const [historyData, setHistoryData] = React.useState<Payment[]>([]);
    const [rentType, setRentType] = React.useState<string>('');
    const isFocused = useIsFocused();

    const fetchPaymentHistory = () => {
        mutationGetPaymentHistory.mutate(
            { rentalId },
            {
                onSuccess: resp => {
                    setRentType(resp.data.data[0].shipRentType);
                    setHistoryData(resp.data.data[0].payment.reverse());
                },
                onError: err => {
                    handleAxiosError(err);
                },
            },
        );
    };
    React.useEffect(() => {
        if (isFocused) {
            fetchPaymentHistory();
        }
    }, [isFocused]);
    
    return (
        <View
            flex
            style={{
                backgroundColor: Color.bgColor,
                position: 'relative',
                justifyContent: 'space-between',
            }}
            padding-10
            testID="paymentHistoryScreen">
            <FlatList
                testID="flatlistHistory"
                ItemSeparatorComponent={() => <View />}
                data={historyData}
                renderItem={({ item, index }) => {
                    if (item.receiptUrl) {
                        return (
                            <HistoryList
                                index={index}
                                item={item}
                                rentType={rentType}
                            />
                        );
                    } else {
                        return <View testID="emptyItem"></View>;
                    }
                }}
                keyExtractor={item => item._id}
            />
        </View>
    );
};

export default PaymentHistory;
