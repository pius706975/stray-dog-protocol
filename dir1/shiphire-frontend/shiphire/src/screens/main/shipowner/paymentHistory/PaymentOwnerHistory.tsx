import React from 'react';
import { PaymentHistoryOwnerProps } from '../../../../types';
import { Color } from '../../../../configs';
import { View } from 'react-native-ui-lib';
import { FlatList } from 'react-native';
import { HistoryList } from './components';

const PaymentOwnerHistory: React.FC<PaymentHistoryOwnerProps> = ({ route }) => {
    const { payment, rentType } = route.params;
    return (
        <View
            flex
            style={{
                backgroundColor: Color.bgColor,
                position: 'relative',
                justifyContent: 'space-between',
            }}
            padding-10
            testID="paymentOwner">
            <FlatList
                testID="flatlistHistory"
                ItemSeparatorComponent={() => <View />}
                data={payment.reverse()}
                renderItem={({ item, index }) => {
                    if (item.receiptUrl) {
                        return (
                            <HistoryList
                                index={index}
                                item={item}
                                rentType={rentType}
                            />
                        );
                    } else {
                        return <View testID="emptyItem"></View>;
                    }
                }}
                keyExtractor={item => item._id}
            />
        </View>
    );
};

export default PaymentOwnerHistory;
