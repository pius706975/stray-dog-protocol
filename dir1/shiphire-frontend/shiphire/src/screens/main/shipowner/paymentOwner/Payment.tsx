import React from 'react';
import { View } from 'react-native-ui-lib';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import { useAcceptPayment } from '../../../../hooks';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import { PaymentOwnerProps } from '../../../../types';
import { handleAxiosError } from '../../../../utils';
import { PaymentReceiptPreview } from '../paymentReceiptOwner';
import { HistoryCard } from './components';
import { useTranslation } from 'react-i18next';

const Payment: React.FC<PaymentOwnerProps> = ({ route, navigation }) => {
    const {
        rentalId,
        paymentReceiptUrl,
        sailingStatus,
        beforeSailingPictures,
        payment,
        rentType,
        lastStatus
    } = route.params;

    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;
    const mutationAcceptPayment = useAcceptPayment();
    const { t } = useTranslation('common')

    const handleSendReceipt = () => {
        dispatch(showProgressIndicator());
        // ! Commented out because of the new flow with the ship tracking history
        // if (beforeSailingPictures.length < 1) {
        // setTimeout(() => {
        //     dispatch(
        //         showModal({
        //             status: 'failed',
        //             text: t('AcceptPaymentStats.textUploadPictures'),
        //         }),
        //     );
        // }, 1000);
        // setTimeout(() => {
        //     navigation.navigate('ShipOwnerTransactionDetail', {
        //         status: 'payment',
        //         rentalId: rentalId,
        //     });
        //     dispatch(hideModal());
        //     dispatch(hideProgressIndicator());
        // }, 4000);
        // return;
        // }
        mutationAcceptPayment.mutate(
            { rentalId },
            {
                onSuccess: () => {
                    dispatch(
                        showModal({
                            status: 'success',
                            text: t('AcceptPaymentStats.textSuccessAcceptPayment'),
                        }),
                    );
                    setTimeout(() => {
                        // ! Commented out because of the new flow with the ship tracking history
                        // navigation.navigate('OwnerTransactionTabNav');
                        navigation.navigate('ShipOwnerTransactionDetail', {
                            status: 'payment',
                            rentalId: rentalId,
                        });
                        dispatch(hideModal());
                        dispatch(hideProgressIndicator());
                    }, 3000);
                },
                onError: err => {
                    dispatch(
                        showModal({
                            status: 'failed',
                            text: t('AcceptPaymentStats.textFailedAcceptPayment'),
                        }),
                    );
                    setTimeout(() => {
                        dispatch(hideModal());
                        dispatch(hideProgressIndicator());
                    }, 3000);
                    handleAxiosError(err);
                },
            },
        );
    };
    console.log(rentType);

    return (
        <ScreenLayout
            testId="PaymentOwnerScreen"
            backgroundColor="light"
            padding={10}>
            <HistoryCard
                navigation={navigation}
                payment={payment}
                rentType={rentType}
            />

            <PaymentReceiptPreview url={paymentReceiptUrl as string} />

            <View
                style={{
                    position: 'absolute',
                    bottom: 30,
                    right: 20,
                }}>
                {lastStatus !== 'payment 3' && (
                    <Button
                        title={t('AcceptPaymentStats.btnAcceptPayment')}
                        color="success"
                        onSubmit={handleSendReceipt}
                    />
                )}
            </View>
        </ScreenLayout>
    );
};

export default Payment;
