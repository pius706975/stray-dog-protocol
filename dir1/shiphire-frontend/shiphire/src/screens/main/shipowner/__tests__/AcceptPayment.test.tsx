import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MainOwnerStackParamList } from '../../../../types';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer, useNavigation } from '@react-navigation/native';
import { render, fireEvent, waitFor, act } from '@testing-library/react-native';
import { Payment } from '../paymentOwner';
import { PaymentOwnerHistory } from '../paymentHistory';

const queryClient = new QueryClient();
const MainOwnerStack = createNativeStackNavigator<MainOwnerStackParamList>();

const mockAdapter = new MockAdapter(httpRequest);
const mockParams = {
    paymentReceiptUrl:
        'https://storage.googleapis.com/shiphire-fdb0e.appspot.com/SH-09212023-63c96de3%20payment%20receipt...',
    rentalId: 'SH-09212023-63c96de3',
    sailingStatus: 'beforeSailing',
    beforeSailingPictures: [
        {
            documentName: 'Image Before Tes',
            documentUrl:
                'https://storage.googleapis.com/download/storage/v1/b/shiphire-fdb0e.appspot.com/o/Image%2',
            description: 'Tes',
        },
    ],
    payment: [
        {
            _id: '659634e1d7911740a0d61b41',
            paymentReminded: [],
            createdAt: new Date('2024-01-04T04:33:12.268Z'),
            paymentApproved: true,
            paymentId: 'PY-01042024-a3029559',
            receiptUrl: 'https://storage.googleapis.com',
        },
        {
            createdAt: new Date('2024-01-04T04:42:33.655Z'),
            _id: '659635bad7911740a0d61caa',
            paymentReminded: [],
            paymentApproved: false,
            paymentId: 'PY-01042024-bf706fd0',
            receiptUrl: 'https://storage.googleapis.com',
        },
        {
            createdAt: new Date('2024-01-04T04:42:33.655Z'),
            _id: '659635bad7911740a0d61cab',
            paymentReminded: [],
            paymentApproved: false,
            paymentId: 'PY-01042024-bf706fd0',
        },
    ],
};

const mockParamsEmptyBeforeSailingPictures = {
    paymentReceiptUrl:
        'https://storage.googleapis.com/shiphire-fdb0e.appspot.com/SH-09212023-63c96de3%20payment%20receipt...',
    rentalId: 'SH-09212023-63c96de3',
    sailingStatus: 'beforeSailing',
    beforeSailingPictures: [],
};

const acceptPaymentMockComponent = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwnerStack.Navigator>
                    <MainOwnerStack.Screen
                        name="Payment"
                        component={Payment}
                        initialParams={mockParams}
                    />
                    <MainOwnerStack.Screen
                        name="PaymentOwnerHistory"
                        component={PaymentOwnerHistory}
                        initialParams={{ payment: mockParams.payment }}
                    />
                </MainOwnerStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

const acceptPaymentMockComponentEmptyBeforeSailingPictures = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainOwnerStack.Navigator>
                    <MainOwnerStack.Screen
                        name="Payment"
                        component={Payment}
                        initialParams={mockParamsEmptyBeforeSailingPictures}
                    />
                </MainOwnerStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

describe('Testing Accept Payment Screen', () => {
    describe('Snapshot Testing', () => {
        it('should render accept payment screen correctly', () => {
            const acceptPaymentScreen = render(acceptPaymentMockComponent);
            expect(acceptPaymentScreen).toMatchSnapshot();
        });
    });
    describe('Unit Testing', () => {
        it('should handle if user has not uploaded before sailing pictures', async () => {
            const { getByText } = render(
                acceptPaymentMockComponentEmptyBeforeSailingPictures,
            );

            fireEvent.press(getByText('Accept payment'));

            await waitFor(() => {
                const modalState = store.getState().modal;
                jest.advanceTimersByTime(1000);
                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'Please upload before sailing pictures first before accepting payment.',
                });
                jest.advanceTimersByTime(4000);

                const progressState = store.getState().progressIndicator;

                expect(progressState).toEqual({
                    visible: false,
                });
            });
        });
        it('should handle accept payment successfully', async () => {
            const { getByText } = render(acceptPaymentMockComponent);

            fireEvent.press(getByText('Accept payment'));
            mockAdapter
                .onPost('/ship-owner/approve-payment')
                .reply(200, { status: 'success' });

            await waitFor(() => {
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: 'Payment has been accepted',
                });
                jest.advanceTimersByTime(3000);

                const progressState = store.getState().progressIndicator;

                expect(progressState).toEqual({
                    visible: false,
                });
            });
        });
        it('should handle accept payment failed', async () => {
            const { getByText } = render(acceptPaymentMockComponent);

            fireEvent.press(getByText('Accept payment'));
            mockAdapter
                .onPost('/ship-owner/approve-payment')
                .reply(500, { status: 'failed' });

            await waitFor(() => {
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'Failed to accept the payment. Please try again.',
                });
                jest.advanceTimersByTime(3000);

                const progressState = store.getState().progressIndicator;

                expect(progressState).toEqual({
                    visible: false,
                });
            });
        });
        it('shoud navigate to payment history when press history button', () => {
            const { getByTestId } = render(acceptPaymentMockComponent);
            const historyButton = getByTestId('historyButton');
            act(() => {
                fireEvent.press(historyButton);
            });
            expect(getByTestId('paymentOwner')).toBeDefined();
        });
    });
});
