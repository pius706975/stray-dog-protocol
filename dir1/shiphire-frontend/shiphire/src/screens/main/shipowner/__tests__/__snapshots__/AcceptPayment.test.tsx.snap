// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Accept Payment Screen Snapshot Testing should render accept payment screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="PaymentOwnerScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  padding-10={true}
                  paddingL-15={true}
                  row={true}
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "padding": 10,
                        "paddingLeft": 15,
                      },
                      {},
                      {
                        "flexDirection": "row",
                      },
                      {
                        "backgroundColor": "#dbdee0",
                        "borderRadius": 5,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "width": "70%",
                        },
                      ]
                    }
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#216dab",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 14,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      PaymentHistory.cardDescription
                    </Text>
                  </View>
                  <View
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "width": "30%",
                        },
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#216dab",
                          "borderRadius": 6,
                          "elevation": 2,
                          "flexDirection": "row",
                          "gap": 0,
                          "justifyContent": "center",
                          "paddingHorizontal": 10,
                          "paddingVertical": 6,
                        }
                      }
                      testID="historyButton"
                    >
                      <Text
                        numberOfLines={0}
                        style={
                          {
                            "borderBottomWidth": 0,
                            "borderColor": undefined,
                            "color": "#f0f6fb",
                            "fontFamily": "Kanit-Medium",
                            "fontSize": 18,
                            "lineHeight": undefined,
                            "textAlign": undefined,
                          }
                        }
                      >
                        PaymentHistory.buttonHistory
                      </Text>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "position": "absolute",
                              "right": 20,
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      {
                        "flex": 1,
                      },
                      undefined,
                      {},
                      {},
                      {},
                      {
                        "backgroundColor": "#f0f6fb",
                        "justifyContent": "space-between",
                        "position": "relative",
                      },
                    ]
                  }
                  testID="PaymentReceiptOwnerPreviewScreen"
                >
                  <RCTScrollView
                    contentContainerStyle={
                      {
                        "alignItems": undefined,
                        "flex": undefined,
                        "flexGrow": 1,
                        "gap": undefined,
                        "justifyContent": undefined,
                        "marginBottom": undefined,
                        "padding": undefined,
                        "paddingVertical": undefined,
                      }
                    }
                  >
                    <View>
                      <View
                        marginT-10={true}
                        paddingH-10={true}
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "paddingHorizontal": 10,
                            },
                            {
                              "marginTop": 10,
                            },
                            {},
                            {
                              "backgroundColor": "#dbdee0",
                              "borderRadius": 5,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <Image
                          onLoadEnd={[Function]}
                          onLoadStart={[Function]}
                          source={
                            {
                              "uri": "https://storage.googleapis.com/shiphire-fdb0e.appspot.com/SH-09212023-63c96de3%20payment%20receipt...",
                            }
                          }
                          style={
                            {
                              "borderRadius": 5,
                              "height": "90%",
                              "resizeMode": "cover",
                              "width": "100%",
                            }
                          }
                        />
                      </View>
                    </View>
                  </RCTScrollView>
                </View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "bottom": 30,
                        "position": "absolute",
                        "right": 20,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#1fcc44",
                        "borderRadius": 6,
                        "elevation": 2,
                        "flexDirection": "row",
                        "gap": 0,
                        "justifyContent": "center",
                        "paddingHorizontal": 10,
                        "paddingVertical": 6,
                      }
                    }
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#f0f6fb",
                          "fontFamily": "Kanit-Medium",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      Accept payment
                    </Text>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "position": "absolute",
                            "right": 20,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Payment"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
